server/src/test/java/io/crate/integrationtests/CopyIntegrationTest.java

```diff
@@ -705,6 +705,42 @@ public void testCopyWithGeneratedPartitionColumnThatIsPartOfPrimaryKey() throws
         assertThat(response.rowCount(), is(2L));
     }
 
+    @Test
+    public void test_copy_from_return_summary_with_failed_rows_line_numers_limitted_to_50() throws Exception {
+
+        execute("create table tmp_t1 (v integer)");
+        Path tmpDir = newTempDir(LifecycleScope.TEST);
+        Path target = Files.createDirectories(tmpDir.resolve("target"));
+        String fileName = "uzulo.json";
+
+        // two correct inserts
+        tmpFileWithLines(Arrays.asList(
+            "{\"v\": 100}",
+            "{\"v\": 200}"
+        ), fileName, target);
+
+        int numBrokenInserts = 60;
+        String [] brokenInserts = new String[numBrokenInserts];
+        for (int i=0; i < numBrokenInserts; i++) {
+            brokenInserts[i] = String.format("{ \"v\": %d, \"bob\": 42", i); // bob is unexpected
+        }
+        tmpFileWithLines(Arrays.asList(brokenInserts), fileName, target);
+
+        execute("copy tmp_t1 from ? return summary", new Object[]{target.toUri().toString() + "*"});
+
+        assertThat(printedTable(response.rows()),
+                   containsString("mapping set to strict, dynamic introduction of [bob] within [default] is not allowed"));
+
+        assertEquals(response.rows()[1].length, 5);
```

Comment:
Using `assertThat` is prefered over `assertEquals` as it returns nicer errors messages. Also the API has clearer definitions of which argument is expected  and which given.
Looks like using it would also highlight current test failure in a better way ;-)